### PR TITLE
[OSCI] Bug fix: De-alias AST module export

### LINKED ISSUE #1081: [BUG] Fix aliased AST module export
**Describe the bug**

In `src/components/search_bar/search_bar.tsx`, line 40, exporting the module `AST` under the alias `Ast` causes the module to be unrecognized when the code is compiled. This generates a warning in my terminal: 

```
WARNING in /Users/johnathonbowers/Desktop/OpenSearch/oui/src/components/search_bar/index.ts 30:0-49
"export 'Ast' was not found in './search_bar'
 @ /Users/johnathonbowers/Desktop/OpenSearch/oui/src/components/index.js
 @ ./routes.js
 @ ./index.js
```

**To Reproduce**
1. On a Mac, run `yarn install` and `yarn start` to launch the development version of the OUI documentation site.
2. Check the terminal or the browser console to see the warning.

**Expected behavior**

The site should load without compile warnings such as this.

**Screenshots**

Screenshot 2023-10-13 at 11 40 36 AM

Screenshot 2023-10-13 at 11 24 36 AM

**Host/Environment (please complete the following information):**
 - OUI Version: [1.4.0-alpha.1]
 - OSD Version (if applicable): N/A
 - OS: MacOS Ventura 13.5.2
 - Browser and version: Chrome Version 116.0.5845.187 (Official Build) (x86_64)

**Additional context**

Following the discussion in Issue #1063 , this issue can be fixed temporarily by de-aliasing the `AST` export in `search_bar/index.ts` and changing `Ast` to `AST` in this file's dependent(s).

### PR DESCRIPTION
**Pull request by JohnathonBowers**
### Description
These changes resolve the compile warning regarding the `AST` module export in `src/components/search_bar/search_bar.tsx`. The `AST` module export is de-aliased (was formerly aliased as `Ast`), and dependent files are updated to reflect the de-aliasing. 

### Issues Resolved
Fixes #1081 

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] All tests pass
  - [x] `yarn lint`
  - [x] `yarn test-unit`
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/components/search_bar/search_bar.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ import { Query } from './query';
 import { CommonProps } from '../common';
 import { OuiFieldSearchProps } from '../form/field_search';
 
-export { Query, AST as Ast } from './query';
```
Comment (BSFishy) [inline src/components/search_bar/search_bar.tsx:40]: I think changing `Ast` -> `AST` would be a breaking change. Would a separate export directly from the original file work?

Comment (JohnathonBowers) [reply src/components/search_bar/search_bar.tsx]: Thank you for bringing this up! I pushed a new commit reverting my changes just so I can preserve the commit history. I've tried to research this issue further, but I think I may be heading down a rabbit hole. :) Would you suggest tabling this issue for right now? Or do you have any other suggestions for a next step?

Comment (BSFishy) [reply src/components/search_bar/search_bar.tsx]: Sure, feel free to focus on something else. If it's not blocking anything then it doesn't need to be a priority. I would recommend converting it to draft if you're not actively working on it.

Comment (JohnathonBowers) [reply src/components/search_bar/search_bar.tsx]: Sounds good. No, it's not blocking anything. I'll convert this PR to a draft.

Comment (JohnathonBowers) [reply src/components/search_bar/search_bar.tsx]: I've recently updated my MacOS to 14.0 (Sonoma). I don't know if that affected anything, but I'm no longer getting the warning in my console about the `Ast` module not being found when I run the most recent version of the `oui` repo. Do you think that it's fine to go ahead and close this PR, as the issue seems to be resolved?

Comment (BSFishy) [reply src/components/search_bar/search_bar.tsx]: Yeah I think you could close it in that case

Comment (JohnathonBowers) [reply src/components/search_bar/search_bar.tsx]: Sounds good. I'll close it.

**Pull request closed without merging**